tests/unit/test_run.py

```diff
@@ -0,0 +1,1788 @@
+    cols = copy.deepcopy(test_pd_df.columns.values)
+    cols[-3:] = years
+    df.columns = cols
+
+    df = ScmDataFrame(df)
+
+    obs = df._data.index
+    exp = pd.Index(
+        [dt.datetime(2005, 1, 1), dt.datetime(2010, 1, 1), dt.datetime(2015, 1, 1)],
+        name="time",
+        dtype="object",
+    )
+    pd.testing.assert_index_equal(obs, exp)
+
+
+def test_init_with_year_columns(test_pd_df):
+    df = ScmDataFrame(test_pd_df)
+    tdf = get_test_pd_df_with_datetime_columns(test_pd_df)
+    pd.testing.assert_frame_equal(df.timeseries().reset_index(), tdf, check_like=True)
+
+
+def test_init_with_decimal_years():
+    inp_array = [2.0, 1.2, 7.9]
+    d = pd.Series(inp_array, index=[1765.0, 1765.083, 1765.167])
+    cols = {
+        "model": ["a_model"],
+        "scenario": ["a_scenario"],
+        "region": ["World"],
+        "variable": ["Primary Energy"],
+        "unit": ["EJ/yr"],
+    }
+
+    res = ScmDataFrame(d, columns=cols)
+    assert (
+        res["time"].unique()
+        == [
+            dt.datetime(1765, 1, 1, 0, 0),
+            dt.datetime(1765, 1, 31, 7, 4, 48),
+            dt.datetime(1765, 3, 2, 22, 55, 11),
+        ]
+    ).all()
+    npt.assert_array_equal(res._data.loc[:, 0].values, inp_array)
+
+
+def test_init_df_from_timeseries(test_scm_df_mulitple):
+    df = ScmRun(test_scm_df_mulitple.timeseries())
+
+    assert_scmdf_almost_equal(df, test_scm_df_mulitple, check_ts_names=False)
+
+
+def test_init_df_with_extra_col(test_pd_df):
+    tdf = test_pd_df.copy()
+
+    extra_col = "test value"
+    extra_value = "scm_model"
+    tdf[extra_col] = extra_value
+
+    df = ScmDataFrame(tdf)
+
+    tdf = get_test_pd_df_with_datetime_columns(tdf)
+    assert extra_col in df.meta
+    pd.testing.assert_frame_equal(df.timeseries().reset_index(), tdf, check_like=True)
+
+
+def test_init_iam(test_iam_df, test_pd_df):
+    a = ScmDataFrame(test_iam_df)
+    b = ScmDataFrame(test_pd_df)
+
+    assert_scmdf_almost_equal(a, b)
+
+
+def test_init_self(test_iam_df):
+    a = ScmDataFrame(test_iam_df)
+    b = ScmDataFrame(a)
+
+    assert_scmdf_almost_equal(a, b)
+
+
+def test_as_iam(test_iam_df, test_pd_df, iamdf_type):
+    df = ScmDataFrame(test_pd_df).to_iamdataframe()
+
+    # test is skipped by test_iam_df fixture if pyam isn't installed
+    assert isinstance(df, iamdf_type)
+
+    pd.testing.assert_frame_equal(test_iam_df.meta, df.meta)
+    # we switch to time so ensure sensible comparison of columns
+    tdf = df.data.copy()
+    tdf["year"] = tdf["time"].apply(lambda x: x.year)
+    tdf.drop("time", axis="columns", inplace=True)
+    pd.testing.assert_frame_equal(test_iam_df.data, tdf, check_like=True)
+
+
+def test_get_item(test_scm_run):
+    assert test_scm_run["model"].unique() == ["a_iam"]
+
+
+def test_get_item_not_in_meta(test_scm_run):
+    dud_key = 0
+    error_msg = re.escape("I don't know what to do with key: {}".format(dud_key))
```

Comment:
Should we make this error message something like, "[meta_col] is not in metadata" just to be slightly clearer? I've always found the reference to 'key' a bit weird